### ISSUE TITLE
feat: determine available "environments" via directory + glob

### ISSUE THREAD

**Issue by bitsofinfo** [roboll/helmfile#747]
Would be great to be able to do something like this:

```
helmfile --environments-dir /path/to/environments -e dog myhelmfile.yaml apply 
```

where `/path/to/environments` has
```
dog/ 
   1.yaml
   2.yaml
cat/
   1.yaml
   2.yaml
```

And then I can just drop in new environments and not have to maintain some index of them in an "environments:" element.

**Comment by mumoshu** [comment]
Interesting idea! I agree we'd want something like this.

Let's split this into two different problems to discuss tho:

1. A convention to treat each sub-directory a helmfile environment
2. How to specify the parent directory for 1. - the proposal includes a flag `--environments-dir`. do we need it in config files as well?

**Comment by bitsofinfo** [comment]
Regarding item 2, do you mean if a command line argument like `--environments-dir` was passed, would helmfile also merge traditionally declared environments like:

```
environments:
   hog:
      ...
```

I would say yes, if both are present, the total set of environments is the union of the two. However I don't think they would need to be merged (for example having both a environment directory named "dog" and a separately defined "environments.dog" in a helmfile probobly would not need to be supported.

**Comment by mumoshu** [comment]
@bitsofinfo Good point. I don't think the merge functionality isn't a must-have, and yeah, the implementation would be much simpler if it isn't need to be supported!

**Comment by bitsofinfo** [comment]
agreed, my use case doesn't need that. I really just want to be able to drop new "environment" directories with arbitrary values files within them under a `--environments-dir` and have them immediately available. For me an "environment" would be either/or. (i.e. living under `--environments-dir` OR traditionally declared via `environments.[envname]`.... but not across both.
